vermouth/gmx/itp_read.py

```diff
@@ -0,0 +1,533 @@
+        'pairs_nb':2,
+        'position_restraints':1,
+        'distance_restraints':2,
+        'dihedral_restraints':4,
+        'orientation_restraints':2,
+        'angle_restraints':4,
+        'angle_restraints_z':2
+    }
+
+    def __init__(self, force_field):
+
+        super().__init__()
+        self.force_field = force_field
+        self.current_block = None
+        self.current_meta = None
+        self.blocks = collections.OrderedDict()
+
+        self.header_actions = {
+            ('moleculetype', ): self._new_block
+        }
+
+    def dispatch(self, line):
+        """
+        Looks at `line` to see what kind of line it is, and returns either
+        :meth:`parse_header` if `line` is a section header or
+        :meth:`parse_section` otherwise. Calls :meth:`is_section_header` to see
+        whether `line` is a section header or not.
+
+        Parameters
+        ----------
+        line: str
+
+        Returns
+        -------
+        collections.abc.Callable
+            The method that should be used to parse `line`.
+        """
+
+        if self.is_section_header(line):
+            return self.parse_header
+        elif self.is_def(line):
+            return self.parse_def
+        else:
+            return self.parse_section
+
+    def is_def(self, line):
+        """
+        Parameters
+        ----------
+        line: str
+            A line of text.
+
+        Returns
+        -------
+        bool
+            ``True`` iff `line` is a def statement.
+
+        """
+        if line.startswith('#') :
+           return True
+        else:
+           return False
+
+
+    def parse_def(self, line, lineno=0):
+        """
+        Parses the beginning and end of define sections
+        with line number `lineno`. Sets :attr:`current_meta`
+        when applicable. Does check if ifdefs overlap.
+
+        Parameters
+        ----------
+        line: str
+        lineno: str
+
+        Returns
+        -------
+        object
+            The result of calling :meth:`finalize_section`, which is called
+            if a section ends.
+
+        Raises
+        ------
+        IOError
+            If the def sections are missformatted
+        """
+                        
+        if line == '#endif' and self.current_meta != None:
+           self.current_meta=None
+
+        elif line == '#endif' and self.current_meta == None:
+           raise IOError("Your #ifdef section is orderd incorrectly." 
+			 "At line {} I read #endif but I haven not read"
+                         "a ifdef before.".format(lineno)) from error
+
+
+        elif line != '#endif' and self.current_meta == None:
+           condition, tag = line.split()
+           self.current_meta = {'tag':tag,'condition':condition}
+
+        else:
+           raise IOError("Your #ifdef section is orderd incorrectly." 
+			 "At line {} I read #endif but I haven not read"
+                         "a ifdef before.".format(lineno)) from error
```

Comment:
Wrong error message, and same comment as before

Reply:
done